lib/tasks/one_off.rake

```diff
@@ -45,4 +45,86 @@ namespace :pvb do
       vs.update_column(:visitor_id, vs.visit.principal_visitor.id)
     end
   end
+
+  # rubocop:disable Metrics/MethodLength
+  # rubocop:disable Lint/AssignmentInCondition
+  # rubocop:disable Style/RescueModifier
+  def check_prisoner_details(client, number, dob)
+    RequestStore.store[:custom_log_items] = {}
+    log = { noms_id: number }
+
+    begin
+      response = client.get('/lookup/active_offender',
+        noms_id: number,
+        date_of_birth: dob)
+
+      log[:found] = !!response['found']
+    rescue Nomis::APIError => e
+      log[:error] = e.message
+    end
+
+    log[:timing] = RequestStore.store[:custom_log_items][:api]
+    log
+  end
+
+  def new_worker(pool, queue, thread_number)
+    Thread.new do
+      request_id = "check_prisoner_details_#{thread_number}"
+      RequestStore.store[:request_id] = request_id
```

Comment:
The request_id gets sent as a header, and kibana parses it from the gateway logs